src/utilities/googleAnalytics.ts

```diff
@@ -10,6 +10,11 @@ import isBorrowing from './isBorrowing';
 import { v4 as uuidv4 } from 'uuid';
 
 const SESSION_ID = uuidv4();
+let USER_ADDRESS = '';
+
+export const saveUserAddress = (address: string) => {
+  USER_ADDRESS = address;
```

Comment:
is there a cleaner way to cache the user address vs. using a global variable? wdyt @ssbarbee @ioanaroceanu2000 ?

Reply:
the other way is to use the `WalletContext` which can be called only in React components but this requires code change in all components that send a GA event. Filip's opinion might help

Reply:
best way is to include the WALLET address as required param in the pushEvent function. This way we remove the "state" from GA since it if you think about it the requirement states: "publish the user address to GA" (or translated extend the GA utility with param address)

Reply:
if you have places where you cannot use the "useWallet()" you can read from localStorage we should be keeping the address there

Reply:
yeah, i also think we should aim to remove state from ga

Reply:
State is evil... it makes everything  more complex to test and refactor later on

Reply:
8eb29680a912799f4984988c57ad166a8d269a11 commit with proposed param. I basically used the wallet hook wherever the `pushEvent()` function was called.